src/dotnet-grpc/Commands/CommandBase.cs

```diff
@@ -0,0 +1,358 @@
+
+                    if (remoteItem == null)
+                    {
+                        Console.Out.WriteLine($"Warning: Could not find a reference that uses the source url `{reference}`.");
+                        continue;
+                    }
+
+                    resolvedReferences.Add(remoteItem);
+                    continue;
+                }
+
+                var localItem = protobufItems.SingleOrDefault(p => p.UnevaluatedInclude == reference);
+
+                if (localItem == null)
+                {
+                    Console.Out.WriteLine($"Warning: Could not find a reference for the file `{reference}`.");
+                    continue;
+                }
+
+                resolvedReferences.Add(localItem);
+            }
+
+            return resolvedReferences;
+        }
+
+        internal string[] GlobReferences(string[] references)
+        {
+            if (Project == null)
+            {
+                throw new InvalidOperationException("Internal error: Project not set.");
+            }
+
+            var expandedReferences = new List<string>();
+
+            foreach (var reference in references)
+            {
+                if (IsUrl(reference))
+                {
+                    expandedReferences.Add(reference);
+                    continue;
+                }
+
+                if (Path.IsPathRooted(reference))
+                {
+                    var directoryToSearch = Path.GetPathRoot(reference);
+                    var searchPattern = reference.Substring(directoryToSearch.Length);
+
+                    expandedReferences.AddRange(Directory.GetFiles(directoryToSearch, searchPattern));
+                    continue;
+                }
+
+                try
+                {
+                    expandedReferences.AddRange(
+                        Directory.GetFiles(Project.DirectoryPath, reference)
+                            .Select(r => r.Replace(Project.DirectoryPath + Path.DirectorySeparatorChar, string.Empty)));
+                }
+                catch (DirectoryNotFoundException) { }
+            }
+
+            return expandedReferences.ToArray();
+        }
+
+        public void RemoveProtobufReference(ProjectItem protobufRef, bool removeFile)
+        {
+            if (Project == null)
+            {
+                throw new InvalidOperationException("Internal error: Project not set.");
+            }
+
+            Project.RemoveItem(protobufRef);
+
+            if (removeFile)
+            {
+                File.Delete(Path.IsPathRooted(protobufRef.UnevaluatedInclude) ? protobufRef.UnevaluatedInclude : Path.Combine(Project.DirectoryPath, protobufRef.UnevaluatedInclude));
+            }
+        }
+
+        public static bool IsUrl(string reference)
+        {
+            return reference.StartsWith("http") && Uri.TryCreate(reference, UriKind.Absolute, out var _);
```

Comment:
Call TryCreate first and then you could test for http with Uri.Scheme.

https://docs.microsoft.com/en-us/dotnet/api/system.uri.scheme?view=netframework-4.8

Also, StartsWith should ignore culture and be case insensitive. Add a test for `HTTP://LOCALHOST/BLAH.PROTO`